README.md

```diff
@@ -19,6 +19,39 @@ Add the ServiceProvider class to the providers array in `config/app.php`:
 
 From here, we will take care of everything based on your configuration. The agent and the middleware will be registered, and transactions will be send to Elastic.
 
+## Collectors
+The default collectors typically listen on events to measure portions of the request such as framework loading, database queries, or jobs.
+
+The SpanCollector in particular allows you to measure any section of your own code via the `ApmCollector` Facade:
+
+```php
+use AG\ElasticApmLaravel\Facades\ApmCollector;
+
+ApmCollector::startMeasure('my-custom-span', 'custom', 'measure', 'My custom span');
+
+// do something amazing
+
+ApmCollector::stopMeasure('my-custom-span');
+```
+
+To record an additional span around your job execution, you may include the provided job middleware (Laravel 6+ only https://laravel.com/docs/6.x/queues#job-middleware):
+
+```php
+public function middleware()
+{
+    return [
+        app(\AG\ElasticApmLaravel\Jobs\Middleware\RecordTransaction::class),
+    ];
+}
+```
+
+Finally, you can add your own collector to the agent!
```

Comment:
Maybe we should also mention that the instance must implement `AG\ElasticApmLaravel\Collectors\Interfaces\DataCollectorInterface`? And I’m also thinking if we should explain how to measure times within the collector, because it can be tricky.

Reply:
I can do that. I need a bit more direction on the time explanation, just not sure what specifically you'd like explained.

Reply:
I was writing some docs to explain how to implement a custom collector when I realized that start time is relative to the start of the transactions, and we don't make this value available. If a correct start time is not provided, the spans will not be rendered correctly.

We could solve this problem by doing the following:

1. Create a public method in `Agent` class that makes the `$request_start_time` available.
2. Create an abstract class (or even make `TimelineDataCollector` abstract) and ask developers to extend that class that already provides functionality to collect measures. This class will get `$request_start_time` from the Agent, instead of expecting the value as a constructor argument.
3. Add a simple example of how to create a custom collector.

Reply:
I agree on all fronts, I'm working on 1 & 2 right now but also getting the abstract parent to use DI more effectively. Let me know how you like that.

Reply:
That's done. As for 3, not sure how to give an example any simpler than the SpanCollector :joy: Tangentially, I did poke around to find a decent list of [events built-in to Laravel](https://laracasts.com/discuss/channels/general-discussion/where-can-i-get-a-complete-list-of-laravel-5-events?page=1#reply=503332) if you're looking for more ways to expand the package 🤷‍♂.

Anyway, the simplest explanation to me is linking to SpanCollector with a brief description of how it uses startMeasure/stopMeasure and to DBQueryCollector with a brief description of how it uses addMeasure directly.